haanna/haanna.py

```diff
@@ -229,6 +229,13 @@ def get_heating_status(self, root):
                   + log_type+"']/period/measurement"
         return root.find(locator).text == 'on'
 
+    def get_cooling_status(self, root):
+        """Gets the active cooling status"""
+        if self.is_legacy_anna(root):
+            return False # cooling not supported on legacy Anna
+        locator = "appliance[type='heater_central']/logs/point_log[type='cooling_state']/period/measurement"
+        return root.find(locator).text == 'on'
```

Comment:
I'll give it a spin, but I recon we need an `if` or a `try` here - looking at `http://w.x.y.z/core/domain_objects` I don't have `cooling_state` in `heater_central`.

(@bouwew is also working on things, like domestic_hot_water, taking example from that to indicate if-fix)

```
        if root.find(locator):
          return root.find(locator).text == 'on'
        else:
          return None
```

Reply:
Ah right. The root.find will return None.
Will fix that one